lib/puppet/util/windows/file.rb

```diff
@@ -89,10 +89,31 @@ def symlink(target, symlink)
   end
   module_function :symlink
 
+
+  def file_exist(file_name)
+    path = file_name.to_s
```

Comment:
Why do you say `WatchedFile` would be lost?  It's using the `file_system/windows.rb` implementation of stat.